akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1226,7 +1223,7 @@ private[stream] object Collect {
           val holder = buffer.dequeue()
           holder.elem match {
             case Success(elem) ⇒
-              push(out, elem)
+              if (elem != null) push(out, elem)
```

Comment:
Stage will not complete if null element is emmited last. `pushNextIfPossible()` should be added in case of `null`. I will add a test case for this